Automattic/simplenote-android
# Improvements to collaboration UI

Fixes #1508

### Fix
This PR adds a series of improvements asked in #1508. Those improvements include:

1. Update the Collaborator UI to use a FAB to add new collaborators. This is more consistent with other parts of the UI (e.g. tags)

| Empty List      | Collaborators List |
| ----------- | ----------- |
|  ![image](https://user-images.githubusercontent.com/195721/134980512-ec2485f6-da34-4fc5-a29a-c81e15a98943.png)    |    ![image](https://user-images.githubusercontent.com/195721/135012299-030e5efd-1df0-447d-ae1d-7169d1fbe863.png)   |




2. Validate whether the user is trying to add their own email as a collaborator

![image](https://user-images.githubusercontent.com/195721/134980732-5c4d1eae-e74d-4a0d-8bb7-0af36ecd7a89.png)


### Test

1. Go to a note
2. Tap on the menu on the top right corner (dot ...)
3. Tap on collaborators
4. Tap on Add Collaborator and type anything that is not a valid email. A message invalid collaborator should be shown.
5. Write a valid email address and tap on  `Accept`. Should update the layout to list of collaborators
6. Tap on the cross icon to remove the collaborator. Should update the layout to show the initial message
7. Tap on Add Collaborator and enter an email address from another valid Simplenote account
8. The note should be shared with the other account
9. Tap on Add Collaborator and enter your account's email. It should show the error message: `You cannot add yourself as a collaborator`

### Review

Only one developer and one designer are required to review these changes, but anyone can perform the review.


### Release

These changes do not require release notes.
